python/paddle/fluid/clip.py

```diff
@@ -271,7 +271,7 @@ def _process_context(self, context, param, grad):
                     "All parameters' 'clip_norm' of a same group should be the same"
                 )
 
-        square = grad * grad
+        square = layers.square(grad)
```

Comment:
In some models, the grad is sparse, but the `sparse * sparse` is meaningless, so I replace with `square`. here.